File: content/problems/2839-check-if-strings-can-be-made-equal-with-operations-i/solutions/compare-character-counts-by-index-parity/Solution.kt
class Solution {
    fun canBeEqual(s1:String,s2:String):Boolean{
        val count=Array(2){
            IntArray(26)
        };
        for(i in 0 until 4){
            count[i%2][s1[i]-'a']++;
            count[i%2][s2[i]-'a']--
        };
        return count.all{
            row->row.all{
                it==0
            }
        }
    }
}

File: content/problems/2839-check-if-strings-can-be-made-equal-with-operations-i/solutions/compare-character-counts-by-index-parity/solution.cpp
class Solution {
    public:bool canBeEqual(string s1,string s2){
        int count[2][26]={
        };
        for(int i=0;i<4;++i){
            ++count[i%2][s1[i]-'a'];
            --count[i%2][s2[i]-'a'];
        }
        for(auto& row:count)for(int value:row)if(value)return false;
        return true;
    }
};
